lib/util/createHash.js

```diff
@@ -4,21 +4,51 @@
 */
 "use strict";
 
-/** @typedef {{new(): Hash}} HashConstructor */
-/**
- * @typedef {Object} Hash
- * @property {function(string|Buffer, string=): Hash} update
- * @property {function(string): string} digest
- */
-
 const BULK_SIZE = 1000;
 
-class BulkUpdateDecorator {
+class BaseHash {
+	/**
+	 * Update hash {@link https://nodejs.org/api/crypto.html#crypto_hash_update_data_inputencoding}
+	 * @param {string|Buffer} data data
+	 * @param {string} [inputEncoding] data encoding
```

Comment:
```suggestion
	 * @param {string=} inputEncoding data encoding
```

we prefer this style in the code base

Reply:
done